spicepy/_client.py

```diff
@@ -84,6 +113,18 @@ def _threaded_flight_do_get(self, ticket: Ticket):
 
         return thread.reader
 
+    def _threaded_firecache_do_get(self, ticket: Ticket):
+        thread = _ArrowFlightCallThread(
+            ticket=ticket,
+            flight_options=self._firecache_options,
+            flight_client=self._firecache_client
+        )
+        thread.start()
+        while thread.is_alive():
+            thread.join(1)
+
+        return thread.reader
+
```

Comment:
this (and other code sections) look identical to the `_flight` variant, except for a different `_options` param.

I'd be inclined to factor this a little more. One possibility is pushing this logic into a new `Flight` object, and you create 2 instantiations, one with normal flight_endpoint and flight_options and the other with the firecache values, then all the grpc+flight logic will be common, which seems to make sense from the current code.

Reply:
Nice! 🔥